src/en/sunshinebutterflyscans/src/eu/kanade/tachiyomi/extension/en/sunshinebutterflyscans/SunshineButterflyScans.kt

```diff
@@ -1,5 +1,244 @@
 package eu.kanade.tachiyomi.extension.en.sunshinebutterflyscans
 
-import eu.kanade.tachiyomi.multisrc.madara.Madara
+import android.util.Base64
+import eu.kanade.tachiyomi.lib.cryptoaes.CryptoAES
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.interceptor.rateLimit
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import okhttp3.Response
+import rx.Observable
+import uy.kohesive.injekt.injectLazy
 
-class SunshineButterflyScans : Madara("Sunshine Butterfly Scans", "https://sunshinebutterflyscan.com", "en")
+class SunshineButterflyScans : HttpSource() {
+
+    override val name = "Sunshine Butterfly Scans"
+
+    override val baseUrl = "https://wings.sbs"
+    private val cdnUrl = "$baseUrl/images/projcoverjpeg/"
+
+    override val lang = "en"
+
+    override val supportsLatest = true
+
+    // Madara -> custom theme
+    override val versionId = 2
+
+    override val client = network.cloudflareClient.newBuilder()
+        .rateLimit(2)
+        .build()
+
+    override fun headersBuilder() = super.headersBuilder()
+        .add("Referer", "$baseUrl/")
+
+    private fun apiHeadersBuilder() = headersBuilder().apply {
+        add("Accept", "*/*")
+        add("Host", baseUrl.toHttpUrl().host)
+    }
+
+    private val apiHeaders by lazy { apiHeadersBuilder().build() }
+
+    private val json: Json by injectLazy()
+
+    private val chaptersData by lazy {
+        client.newCall(
+            GET("$baseUrl/json/chapters.json", apiHeaders),
+        ).execute().parseAs<List<Dto>>().groupBy {
+            it.series
+        }.values.map { it.sortedByDescending { it.num } }
+    }
+
+    // ============================== Popular ===============================
+
+    override fun fetchPopularManga(page: Int): Observable<MangasPage> {
+        val mangaList = chaptersData.sortedBy {
+            it.first().series
+        }.map {
+            it.first().toSManga(cdnUrl)
+        }
+
+        return Observable.just(MangasPage(mangaList, false))
```

Comment:
if there are more than a few manga on the site then consider paginating it client side, since app takes quite a while for mapping manga to db in browse

Reply:
There's only 30 manga in total, I don't think its gonna be a problem